File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/quiz/QuizAdminService.java
package ma.zs.zyn.service.facade.admin.quiz;

import java.util.List;
import ma.zs.zyn.bean.core.quiz.Quiz;
import ma.zs.zyn.dao.criteria.core.quiz.QuizCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface QuizAdminService extends  IService<Quiz,QuizCriteria>  {

    List<Quiz> findBySectionId(Long id);
    int deleteBySectionId(Long id);
    long countBySectionCode(String code);




}

File: frontend/src/app/shared/service/admin/quiz/QuizAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {QuizDto} from '../../../model/quiz/Quiz.model';
import {QuizCriteria} from '../../../criteria/quiz/QuizCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class QuizAdminService extends AbstractService<QuizDto, QuizCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/quiz/';
    }

    public constrcutDto(): QuizDto {
        return new QuizDto();
    }

    public constrcutCriteria(): QuizCriteria {
        return new QuizCriteria();
    }
}
